packages/react-utilities/src/utils/getNativeElementProps.ts

```diff
@@ -65,3 +65,26 @@ export function getNativeElementProps<TAttributes extends React.HTMLAttributes<a
 
   return getNativeProps(props, allowedPropNames, excludedPropNames);
 }
+
+/**
+ * Returns the native props that always go to the root slot.
+ *
+ * This function is only for use with components where `root` is *not* the primary slot.
+ * Most components should use {@link getNativeElementProps} for their root slot if it is the primary slot.
+ */
+export const getRootSlotNativeProps = ({
+  style,
+  className,
+}: Pick<React.HTMLAttributes<HTMLElement>, 'style' | 'className'>) => ({ style, className });
+
+/**
+ * Returns the native props that go to the primary slot.
+ *
+ * This function is only for use with components where `root` is *not* the primary slot.
+ * Most components should use {@link getNativeElementProps} for their root slot if it is the primary slot.
+ */
+export const getPrimarySlotNativeProps = (
+  tagName: keyof JSX.IntrinsicElements,
+  props: {},
+  excludedPropNames: string[] = [],
+) => getNativeElementProps(tagName, props, [...excludedPropNames, 'style', 'className']);
```

Comment:
I'm not sure this should be added immediately to `react-utilities`, it's once again early generalization IMO. Also, the names are not specific enough and a simple comment won't suffice, I can see this being used wrongly just because of the name confusion

Reply:
This is implementing RFC #18983, and needs to be shared between multiple components for consistency. It should be in a shared place, and `react-utilities` seemed like the appropriate package because it also has `getNativeElementProps`.